### PR TITLE
Ignore js sourcemap files from boot

### PR DESCRIPTION
**Pull request by zbarbuto**
### Description

See discussion from #256  and #245

#### Related issues

- #255

### Checklist

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

cc @bajtos

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by lehni** [comment]
I haven't been part of the previous discussions in #256, but this does look good to me. Let's see what @bajtos thinks, since he pointed out some problems in that issue.

@zbarbuto don't forget to sign the [CLA](https://cla.strongloop.com/agreements/strongloop/loopback-boot), as otherwise the checks won't pass and we won't be able to merge.

**Comment by zbarbuto** [comment]
Whoops. Forget I need to sign it for each individual SL repo I contribute to.

**Comment by bajtos** [comment]
@slnode ok to test

File: lib/compiler.js [hunk 1]
```diff
@@ -661,6 +661,7 @@ function resolveRelativePaths(relativePaths, appRootDir) {
 function getExcludedExtensions() {
   return {
     '.json': '.json',
+    '.map': '.map',
```
Comment (bajtos) [inline lib/compiler.js]: Please add a comment explaining that this is a temporary workaround - see https://github.com/strongloop/loopback-boot/pull/245#issuecomment-329098622

> I think the `.map` entry should be also accompanied by a comment explaining that it's a partial fix for a bigger problem and pointing to https://github.com/strongloop/loopback-boot/pull/245#issuecomment-311052798

File: test/compiler.test.js [hunk 2]
```diff
@@ -2187,6 +2187,26 @@ describe('compiler', function() {
...
+        'routes': {
+          './my-middleware': {},
+        },
+      });
+
+      var instructions = boot.compile(appdir.PATH);
+
+      expect(instructions.middleware.middleware[0]).have.property(
+        'sourceFile', middleware);
+      expect(instructions.middleware.middleware[0]).not.have.property(
```
Comment (bajtos) [inline test/compiler.test.js]: This second expectation is redundant - if the `sourceFile` property is set to `middleware`, then the second expectation will always pass. If `sourceFile` is not set to `middleware`, then the first `expect` will abort the test and this second expectation won't be run at all.

Please remove these two lines (2206-2207).

Comment (zbarbuto) [reply test/compiler.test.js]: Ah, yes, fair point. My goal was to ensure that the map file is definitely never loaded but I missed the index being used. Will remove.

File: lib/compiler.js (hunk 1 above)
Comment (zbarbuto) [reply lib/compiler.js]: Ah, yep, missed that. Cheers.

**Comment by bajtos** [comment]
Released in `loopback-boot@2.26.2`, thank you for the contribution!

**Pull request merged**